File: java_to_c++/dog.cpp
#include <iostream>
#include "dog.h"

using namespace std;

Dog::Dog()
{
	age = 0;
}

string Dog::bark()
{
	return name + " says ARF!";
}

string Dog::bark (string something)
{
	return name + " says " + something;
}

void Dog::setName (string name)
{
	this->name = name;
}

string Dog::getName()
{
	return name;
}

Dog::~Dog()
{
	cout << "Good bye cruel world!" << endl;
}

File: java_to_c++/Dog.java
public class Dog
{
	private boolean sex;
	private String name;
	private int age;
	
	public Dog()
	{
		age = 0;
		name = "UNKNOWN";
	}
	
	public String bark()
	{
		return name + " says ARF!";
	}
	
	public String bark (String something)
	{
		return name + " says " + something;
	}
	
	public void setName (String name)
	{
		this.name = name;
	}
	
	public String getName()
	{
		return name;
	}

	public static void main (String [] args)
	{
		Dog d = new Dog();
		d.setName("Fido");
		System.out.println(d.bark());
	}
}
